aptos-move/framework-experimental/tests/move_unit_tests.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+// This is duplicated from `aptos-move/move-examples/tests/move_unit_tests.rs` then slightly modified.
+
+use aptos_framework::extended_checks;
+use aptos_gas_schedule::{MiscGasParameters, NativeGasParameters, LATEST_GAS_FEATURE_VERSION};
+use aptos_types::{
+    account_address::AccountAddress,
+    on_chain_config::{aptos_test_feature_flags_genesis, Features, TimedFeaturesBuilder},
+};
+use aptos_vm::natives;
+use move_cli::base::test::{run_move_unit_tests, UnitTestResult};
+use move_package::{source_package::std_lib::StdVersion, CompilerConfig};
+use move_unit_test::UnitTestingConfig;
+use move_vm_runtime::native_functions::NativeFunctionTable;
+use std::{collections::BTreeMap, path::PathBuf};
+use tempfile::tempdir;
+
+pub fn path_in_crate<S>(relative: S) -> PathBuf
+where
+    S: Into<String>,
+{
+    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    path.push(relative.into());
+    path
+}
+
+/// Get the local framework path based on this source file's location.
+/// Note: If this source file is moved to a different location, this function
+/// may need to be updated.
+fn get_local_framework_path() -> String {
+    PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .parent()
+        .map(|p| p.join("framework"))
+        .expect("framework path")
+        .to_string_lossy()
+        .to_string()
+}
+
+pub fn aptos_test_natives() -> NativeFunctionTable {
```

Comment:
Is this copied over from https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/tests/move_unit_test.rs#L54?

Reply:
yes. It's a small function, so duplication is fine?
If we want to reuse, I'm sure which crate this func should live in.

Reply:
Well I see that you're already importing the `aptos-framework` crate, so maybe just keep it there?

Reply:
Actually, `framework-experimental` is intended to be barely some Move modules... (and that's why I initially tend to copy those util functions over)

Having to depend on `aptos-framework` just to be able to run UTs sounds weird, but i'm not sure what I missed.

Currently, the only `aptos-framework` dependency is the `extended_checks`.

Reply:
Looking more, why does the `ExtendedChecker` live in `aptos-framework`?

@vgao1996 does it make sense to you?

Reply:
Personally I'm less concerned about duplicating `fn run_tests_for_pkg` but I'd like to keep only 1 version of `fn aptos_test_natives()` so we don't get confused. We could consider moving all framework natives into a separate crate but I guess that can be done some other day.

> why does the ExtendedChecker live in aptos-framework? Seems more like a general Move thing?

Good question. I guess it's because many of these checks that are specific to Aptos at the framework level, such as resource groups.

Reply:
looking more and here's something weird...

`fn aptos_test_natives()` needs:
- `aptos_framework::extended_checks`
- `aptos_vm::natives::configure_for_unit_test`

It means you can't put it in `aptos_framework` since `aptos_vm` depends on `aptos_framework`.
Currently it lives in the `tests` folder `aptos_framework` fine, which is a grey area where the normal crate dependency rule doesn't apply?

Reply:
i guess that's why `move-examples` also has its own `aptos_test_natives`.